### PR TITLE
[AIRFLOW-1883] Get File Size for objects in Google Cloud Storage

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1883

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Added `get_size()` method to `gcs_hook` to get the file size for objects in Google Cloud Storage.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Google doesn't provide local unit test for GCS.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by kaxil** [comment]
cc @criccomini

**Comment by Fokko** [comment]
Thanks @kaxil. Could you write a test for this? I.e. mocking the request.

**Comment by kaxil** [comment]
@Fokko I have added the mock test [here](https://github.com/apache/incubator-airflow/pull/2840/commits/884c02d45f1e66691324a067a2b503831a6477aa) Can you please verify if this is what you wanted? As you will need an actual gcs connection to have a unit test.

**Comment by kaxil** [comment]
@Fokko I am afraid I would not know to write a mock test for this as there needs to be a Google Cloud project for running a test. I also checked the tests written [here](https://github.com/apache/incubator-airflow/blob/master/tests/contrib/hooks/test_gcp_dataproc_hook.py) but I don't know where this tests are executed. I have written a similar test below if this is what you like. Can you confirm?

```python
import unittest
from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook

try:
    from unittest import mock
except ImportError:
    try:
        import mock
    except ImportError:
        mock = None

BUCKET = 'test-bucket' 
OBJECT = 'test-object'

BASE_STRING = 'airflow.contrib.hooks.gcp_api_base_hook.{}'
CLOUDSTORAGE_STRING = 'airflow.contrib.hooks.gcs_hook.{}'


def mock_init(self, gcp_conn_id, delegate_to=None):
    pass


class GoogleCloudStorageHookTest(unittest.TestCase):
    def setUp(self):
        with mock.patch(BASE_STRING.format('GoogleCloudBaseHook.__init__'),
                        new=mock_init):
            self.gcp_gcs_hook = GoogleCloudStorageHook()

    def test_submit(self, gcs_mock):
      with mock.patch(CLOUDSTORAGE_STRING.format('GoogleCloudStorageHook.get_conn', return_value=None)):
        self.gcp_gcs_hook.submit(BUCKET, OBJECT)
        gcs_mock.assert_called_once_with(mock.ANY, BUCKET, OBJECT)
```

File: airflow/contrib/hooks/gcs_hook.py [hunk 1]
```diff
@@ -269,3 +269,32 @@ def list(self, bucket, versions=None, maxResults=None, prefix=None, delimiter=No
...
+        service = self.get_conn()
+        try:
+            response = service.objects().get(
+                bucket=bucket,
+                object=object
+            ).execute()
+
+            if 'name' in response and response['name'][-1] != '/':
+                # Remove Directories & Just check size of files
+                if 'size' in response:
```
Comment (criccomini) [inline airflow/contrib/hooks/gcs_hook.py]: This method seems to return nothing if size is not in response. What's the expected behavior here?

File: airflow/contrib/hooks/gcs_hook.py [hunk 2]
```diff
@@ -269,3 +269,32 @@ def list(self, bucket, versions=None, maxResults=None, prefix=None, delimiter=No
...
+                    size = response['size']
+                    self.log.info('Checking the file size of %s', object)
```
Comment (criccomini) [inline airflow/contrib/hooks/gcs_hook.py]: This log line should be way up before the `service.objects.get()` call

File: airflow/contrib/hooks/gcs_hook.py [hunk 3]
```diff
@@ -269,3 +269,32 @@ def list(self, bucket, versions=None, maxResults=None, prefix=None, delimiter=No
...
+        :type object: string
+        """
+        self.log.info('Checking the file size of %s', object)
+        service = self.get_conn()
+        try:
+            response = service.objects().get(
+                bucket=bucket,
+                object=object
+            ).execute()
+
```
Comment (criccomini) [inline airflow/contrib/hooks/gcs_hook.py:289]: Nit: only one newline here.

File: airflow/contrib/hooks/gcs_hook.py [hunk 4]
```diff
@@ -269,3 +269,32 @@ def list(self, bucket, versions=None, maxResults=None, prefix=None, delimiter=No
...
+    def get_size(self, bucket, object):
+        """
+        Gets the size of a file in Google Cloud Storage.
+        :param bucket: The Google cloud storage bucket where the object is.
+        :type bucket: string
+        :param object: The name of the object to check in the Google cloud
+            storage bucket.
+        :type object: string
+        """
+        self.log.info('Checking the file size of %s', object)
```
Comment (criccomini) [inline airflow/contrib/hooks/gcs_hook.py]: Nit: include bucket as well?

**Comment by criccomini** [comment]
Couple of nits. If you clean them up, I can merge.

**Comment by kaxil** [comment]
@criccomini How about now?

**Comment by criccomini** [comment]
LGTM merged

**Pull request closed without merging**